### ISSUE TITLE
vault-unseal-keys secret is not being created

### ISSUE THREAD

**Issue by TomaszKlosinski** [banzaicloud/bank-vaults#1052]
**Describe the bug**:
The secret vault-unseal-keys is not being created when creating a vault instance using bank-vaults operator.

**Expected behaviour**:
Deploying a Vault instance should create vault-unseal-keys secret.

**Steps to reproduce the bug**:
I have a shell script `vault-install.sh` in which I'm using `envsubst` to template the Kubernetes manifests:

```
# Add banzaicloud helm repo
helm repo add banzaicloud-stable https://kubernetes-charts.banzaicloud.com
helm repo update


# Install vault-operator
helm install --namespace=system-vault-operator vault-operator banzaicloud-stable/vault-operator --version 1.3.5 --wait

# Install vault-secrets-webhook
helm install --namespace system-vault-operator vault-secrets-webhook banzaicloud-stable/vault-secrets-webhook --values vault-secrets-webhook-values.yaml --version 1.3.9 --wait


# Create central Vault instance
export NAMESPACE="system-vault-central"
export INSTANCE_NAME="vault-central"
export VAULT_ACCESS_NAMESPACES='["*"]'
cat templates/rbac.yaml | envsubst | kubectl apply -n ${NAMESPACE} -f -
cat templates/cr.yaml | envsubst | kubectl apply -n ${NAMESPACE} -f -
```

Here is the `templates/rbac.yaml` manifest:
```
---
kind: ServiceAccount
apiVersion: v1
metadata:
  name: vault

---

kind: Role
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: vault-secrets
rules:
  - apiGroups:
      - ""
    resources:
      - secrets
    verbs:
      - "*"

---

kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: vault-secrets
roleRef:
  kind: Role
  name: vault-secrets
  apiGroup: rbac.authorization.k8s.io
subjects:
  - kind: ServiceAccount
    name: vault

---

# This binding allows the deployed Vault instance to authenticate clients
# through Kubernetes ServiceAccounts (if configured so).
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: vault-auth-delegator-${NAMESPACE}
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: system:auth-delegator
subjects:
  - kind: ServiceAccount
    name: vault
    namespace: ${NAMESPACE}
```

Here is the `templates/cr.yaml` manifest:
```
apiVersion: "vault.banzaicloud.com/v1alpha1"
kind: "Vault"
metadata:
  name: "${INSTANCE_NAME}"
spec:
  size: 1
  image: vault:1.4.3
  bankVaultsImage: banzaicloud/bank-vaults:latest

  annotations:
    common/annotation: "true"

  vaultAnnotations:
    type/instance: "vault"

  vaultConfigurerAnnotations:
    type/instance: "vaultconfigurer"

  vaultLabels:
    example.com/log-format: "json"

  vaultConfigurerLabels:
    example.com/log-format: "string"

  serviceAccount: vault

  serviceType: ClusterIP

  ingress:
    annotations:
    spec: {}

  volumes:
    - name: vault-file
      persistentVolumeClaim:
        claimName: vault-file

  volumeMounts:
    - name: vault-file
      mountPath: /vault/file

  caNamespaces: ${VAULT_ACCESS_NAMESPACES}

  unsealConfig:
    options:
      preFlightChecks: true
    kubernetes:
      secretNamespace: ${NAMESPACE}

  config:
    storage:
      file:
        path: "${ .Env.VAULT_STORAGE_FILE }"
    listener:
      tcp:
        address: "0.0.0.0:8200"
        tls_cert_file: /vault/tls/server.crt
        tls_key_file: /vault/tls/server.key
    telemetry:
      statsd_address: localhost:9125
    ui: true

  externalConfig:
    policies:
      - name: allow_secrets
        rules: path "secret/*" {
          capabilities = ["create", "read", "update", "delete", "list"]
          }
      - name: readonly_secrets
        rules: path "secret/*" {
                capabilities = ["read", "list"]
              }

    auth:
      - type: kubernetes
        roles:
          - name: default
            bound_service_account_names: ["default", "vault-secrets-webhook", "vault"]
            bound_service_account_namespaces: ['*']
            policies: ["allow_secrets"]
            ttl: 1h

      - type: ldap
        description: LDAP directory authentication.
        options:
          listing_visibility: "unauth"
        config:
          url: ldap://ldap.myserver.xyz:389
          binddn: "cn=binduser,ou=auth,dc=banana,dc=xyz"
          bindpass: ${LDAP_BIND_PASSWORD}
          userdn: "ou=people,dc=banana,dc=xyz"
          userattr: uid
          groupdn: "ou=roles,ou=vault-k8s-${CLUSTER_IDENTIFIER},ou=k8s-vault,dc=applications,dc=banana,dc=xyz"
        groups:
          superusers:
            policies:
              - allow_secrets
          developers:
            policies:
              - readonly_secrets

    secrets:
      - path: secret
        type: kv
        description: General secrets.
        options:
          version: 2

    startupSecrets:
      - type: kv
        path: secret/data/meta
        data:
          data:
            NAMESPACE: "${NAMESPACE}"

  vaultEnvsConfig:
    - name: VAULT_LOG_LEVEL
      value: debug
    - name: VAULT_STORAGE_FILE
      value: "/vault/file"

  istioEnabled: false

  audit:
    - type: file
      description: "File based audit logging device"
      options:
        file_path: stdout

---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: vault-file
spec:
  storageClassName: freenas-retain
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 1Gi
```

**Environment details**:
- Kubernetes version: `v1.18.2`
- Cloud-provider/provisioner: Installation on-prem with kubeadm.
- bank-vaults version: `banzaicloud/bank-vaults:latest`
- Install method (e.g. helm or static manifests): Helm for operator and webhook + manifests for CR.

**Logs**
Vault instance server:
```
$ kubectl logs -n system-vault-central --all-containers -f vault-central-0
time="2020-07-14T11:10:45Z" level=info msg="initializing vault..."
time="2020-07-14T11:10:45Z" level=info msg="vault metrics exporter enabled: :9091/metrics"
[GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
telemetry.disable_hostname has been set to false. Recommended setting is true for Prometheus to avoid poorly named metrics.
 - using env:	export GIN_MODE=release
 - using code:	gin.SetMode(gin.ReleaseMode)
==> Vault server configuration:

                     Cgo: disabled

              Go Version: go1.13.12
[GIN-debug] GET    /metrics                  --> github.com/gin-gonic/gin.WrapH.func1 (3 handlers)
              Listener 1: tcp (addr: "0.0.0.0:8200", cluster address: "0.0.0.0:8201", max_request_duration: "1m30s", max_request_size: "33554432", tls: "enabled")
               Log Level: debug
                   Mlock: supported: true, enabled: true
           Recovery Mode: false
[GIN-debug] Listening and serving HTTP on :9091
time="2020-07-14T11:10:45Z" level=info msg="vault is already initialized"
time="2020-07-14T11:10:45Z" level=info msg="vault is sealed, unsealing"
time="2020-07-14T11:10:45Z" level=error msg="error unsealing vault: unable to get key 'vault-unseal-0': error getting secret for key 'vault-unseal-0': secrets \"vault-central-unseal-keys\" not found"
time="2020-07-14T11:10:50Z" level=info msg="vault is sealed, unsealing"
                 Storage: file
                 Version: Vault v1.4.3

==> Vault server started! Log data will stream in below:
time="2020-07-14T11:10:50Z" level=error msg="error unsealing vault: unable to get key 'vault-unseal-0': error getting secret for key 'vault-unseal-0': secrets \"vault-central-unseal-keys\" not found"
time="2020-07-14T11:10:55Z" level=info msg="vault is sealed, unsealing"
time="2020-07-14T11:10:55Z" level=error msg="error unsealing vault: unable to get key 'vault-unseal-0': error getting secret for key 'vault-unseal-0': secrets \"vault-central-unseal-keys\" not found"

2020-07-14T11:10:44.656Z [INFO]  proxy environment: http_proxy= https_proxy= no_proxy=
time="2020-07-14T11:11:00Z" level=info msg="vault is sealed, unsealing"
2020-07-14T11:10:44.657Z [WARN]  no `api_addr` value specified in config or in VAULT_API_ADDR; falling back to detection if possible, but this value should be manually set
2020-07-14T11:10:44.657Z [DEBUG] storage.cache: creating LRU cache: size=0
time="2020-07-14T11:11:00Z" level=error msg="error unsealing vault: unable to get key 'vault-unseal-0': error getting secret for key 'vault-unseal-0': secrets \"vault-central-unseal-keys\" not found"
time="2020-07-14T11:11:05Z" level=info msg="vault is sealed, unsealing"
time="2020-07-14T11:11:05Z" level=error msg="error unsealing vault: unable to get key 'vault-unseal-0': error getting secret for key 'vault-unseal-0': secrets \"vault-central-unseal-keys\" not found"
2020-07-14T11:10:44.790Z [DEBUG] cluster listener addresses synthesized: cluster_addresses=[0.0.0.0:8201]
time="2020-07-14T11:11:10Z" level=info msg="vault is sealed, unsealing"
time="2020-07-14T11:11:10Z" level=error msg="error unsealing vault: unable to get key 'vault-unseal-0': error getting secret for key 'vault-unseal-0': secrets \"vault-central-unseal-keys\" not found"
time="2020-07-14T11:11:15Z" level=info msg="vault is sealed, unsealing"
time="2020-07-14T11:11:15Z" level=error msg="error unsealing vault: unable to get key 'vault-unseal-0': error getting secret for key 'vault-unseal-0': secrets \"vault-central-unseal-keys\" not found"
time="2020-07-14T11:11:20Z" level=info msg="vault is sealed, unsealing"
time="2020-07-14T11:11:20Z" level=error msg="error unsealing vault: unable to get key 'vault-unseal-0': error getting secret for key 'vault-unseal-0': secrets \"vault-central-unseal-keys\" not found"
time="2020-07-14T11:11:25Z" level=info msg="vault is sealed, unsealing"
```

Vault instance configurer:
```
⎈ kubectl logs -n system-vault-central --all-containers -f vault-central-configurer-fcf4456f9-nr8b6 | head -n50
[GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
 - using env:	export GIN_MODE=release
 - using code:	gin.SetMode(gin.ReleaseMode)

time="2020-07-14T11:10:49Z" level=info msg="vault metrics exporter enabled: :9091/metrics"
[GIN-debug] GET    /metrics                  --> github.com/gin-gonic/gin.WrapH.func1 (3 handlers)
[GIN-debug] Listening and serving HTTP on :9091
time="2020-07-14T11:10:49Z" level=info msg="applying config file : /config/vault-central-configurer/vault-config.yml"
time="2020-07-14T11:10:49Z" level=info msg="checking if vault is sealed..."
time="2020-07-14T11:10:49Z" level=info msg="watching directory for changes: /config/vault-central-configurer/"
time="2020-07-14T11:10:56Z" level=error msg="error checking if vault is sealed: error checking status: Get \"https://vault-central.system-vault-central:8200/v1/sys/seal-status\": dial tcp 10.121.104.15:8200: connect: connection refused, waiting 5s before trying again..."
time="2020-07-14T11:11:01Z" level=info msg="checking if vault is sealed..."
time="2020-07-14T11:11:08Z" level=error msg="error checking if vault is sealed: error checking status: Get \"https://vault-central.system-vault-central:8200/v1/sys/seal-status\": dial tcp 10.121.104.15:8200: connect: connection refused, waiting 5s before trying again..."
time="2020-07-14T11:11:13Z" level=info msg="checking if vault is sealed..."
time="2020-07-14T11:11:20Z" level=error msg="error checking if vault is sealed: error checking status: Get \"https://vault-central.system-vault-central:8200/v1/sys/seal-status\": dial tcp 10.121.104.15:8200: connect: connection refused, waiting 5s before trying again..."
time="2020-07-14T11:11:25Z" level=info msg="checking if vault is sealed..."
```

Vault Operator:
```
⎈ kubectl logs -n system-vault-operator vault-operator-5fc487b958-6vw6w | head -n50
{"level":"info","ts":1594724978.6201375,"logger":"cmd","msg":"Watched namespace: "}
{"level":"info","ts":1594724979.817266,"logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":":8383"}
{"level":"info","ts":1594724979.9208393,"logger":"cmd","msg":"Registering Components."}
{"level":"info","ts":1594724981.0281322,"logger":"cmd","msg":"Starting the Cmd."}
{"level":"info","ts":1594724981.028515,"logger":"controller-runtime.manager","msg":"starting metrics server","path":"/metrics"}
I0714 11:09:41.028452       1 leaderelection.go:242] attempting to acquire leader lease  system-vault-operator/vault-operator-lock...
I0714 11:09:41.047578       1 leaderelection.go:252] successfully acquired lease system-vault-operator/vault-operator-lock
{"level":"info","ts":1594724981.0477529,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"vault-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1594724981.348871,"logger":"controller-runtime.controller","msg":"Starting Controller","controller":"vault-controller"}
{"level":"info","ts":1594724981.348994,"logger":"controller-runtime.controller","msg":"Starting workers","controller":"vault-controller","worker count":1}
{"level":"info","ts":1594725006.8200665,"logger":"controller_vault","msg":"Reconciling Vault","Request.Namespace":"system-vault-central","Request.Name":"vault-central"}
{"level":"info","ts":1594725008.1180053,"logger":"controller_vault","msg":"TLS CA will be regenerated due to: ","error":"an empty CA was provided"}
{"level":"error","ts":1594725038.4181066,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"vault-controller","request":"system-vault-central/vault-central","error":"failed to distribute CA secret for vault: failed to query current secret for vault: Secret \"vault-central-tls\" not found","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:258\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.17.2/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.17.2/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.17.2/pkg/util/wait/wait.go:88"}
{"level":"info","ts":1594725039.4186666,"logger":"controller_vault","msg":"Reconciling Vault","Request.Namespace":"system-vault-central","Request.Name":"vault-central"}
{"level":"info","ts":1594725045.4068177,"logger":"controller_vault","msg":"Reconciling Vault","Request.Namespace":"system-vault-central","Request.Name":"vault-central"}
{"level":"info","ts":1594725111.897657,"logger":"controller_vault","msg":"Reconciling Vault","Request.Namespace":"system-vault-central","Request.Name":"vault-central"}
{"level":"info","ts":1594725177.1839476,"logger":"controller_vault","msg":"Reconciling Vault","Request.Namespace":"system-vault-central","Request.Name":"vault-central"}
{"level":"info","ts":1594725242.4703119,"logger":"controller_vault","msg":"Reconciling Vault","Request.Namespace":"system-vault-central","Request.Name":"vault-central"}
{"level":"info","ts":1594725307.756628,"logger":"controller_vault","msg":"Reconciling Vault","Request.Namespace":"system-vault-central","Request.Name":"vault-central"}
{"level":"info","ts":1594725373.0429838,"logger":"controller_vault","msg":"Reconciling Vault","Request.Namespace":"system-vault-central","Request.Name":"vault-central"}
{"level":"info","ts":1594725438.3293154,"logger":"controller_vault","msg":"Reconciling Vault","Request.Namespace":"system-vault-central","Request.Name":"vault-central"}
{"level":"info","ts":1594725503.6157725,"logger":"controller_vault","msg":"Reconciling Vault","Request.Namespace":"system-vault-central","Request.Name":"vault-central"}
```

Vault Secrets Webhook:
```
$ kubectl logs -n system-vault-operator vault-secrets-webhook-8885654b-8ksb2
time="2020-07-14T11:09:50Z" level=warning msg="no tracer active" app=vault-secrets-webhook
time="2020-07-14T11:09:50Z" level=warning msg="no tracer active" app=vault-secrets-webhook
time="2020-07-14T11:09:50Z" level=warning msg="no tracer active" app=vault-secrets-webhook
time="2020-07-14T11:09:50Z" level=warning msg="no tracer active" app=vault-secrets-webhook
time="2020-07-14T11:09:50Z" level=info msg="Listening on https://:8443" app=vault-secrets-webhook

$ kubectl logs -n system-vault-operator vault-secrets-webhook-8885654b-kssrc
time="2020-07-14T11:09:50Z" level=warning msg="no tracer active" app=vault-secrets-webhook
time="2020-07-14T11:09:50Z" level=warning msg="no tracer active" app=vault-secrets-webhook
time="2020-07-14T11:09:50Z" level=warning msg="no tracer active" app=vault-secrets-webhook
time="2020-07-14T11:09:50Z" level=warning msg="no tracer active" app=vault-secrets-webhook
time="2020-07-14T11:09:50Z" level=info msg="Listening on https://:8443" app=vault-secrets-webhook
```

**Descriptions**
Vault instance:
```
$ kubectl describe -n system-vault-central pod/vault-central-0
Name:         vault-central-0
Namespace:    system-vault-central
Priority:     0
Node:         worker-prod-01.kube-prod.banana.systems/10.202.1.30
Start Time:   Tue, 14 Jul 2020 13:10:41 +0200
Labels:       app.kubernetes.io/name=vault
              controller-revision-hash=vault-central-648dcf59cf
              example.com/log-format=json
              statefulset.kubernetes.io/pod-name=vault-central-0
              vault_cr=vault-central
Annotations:  cni.projectcalico.org/podIP: 10.122.3.150/32
              cni.projectcalico.org/podIPs: 10.122.3.150/32
              common/annotation: true
              prometheus.io/path: /metrics
              prometheus.io/port: 9102
              prometheus.io/scrape: true
              type/instance: vault
              vault.banzaicloud.io/tls-expiration-date: 2021-07-14T11:10:08Z
Status:       Running
IP:           10.122.3.150
IPs:
  IP:           10.122.3.150
Controlled By:  StatefulSet/vault-central
Init Containers:
  config-templating:
    Container ID:  docker://091b45c3de36c02f345c4ed2781d5a038a623bb9c41b2382f15cf2ee538fbc18
    Image:         banzaicloud/bank-vaults:latest
    Image ID:      docker-pullable://banzaicloud/bank-vaults@sha256:5090ef735a295644aa7753c2f94dd4a65af940f8ae69bbc7828c665dcde20edd
    Port:          <none>
    Host Port:     <none>
    Command:
      template
      -file
      /vault/config/vault.json
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Tue, 14 Jul 2020 13:10:43 +0200
      Finished:     Tue, 14 Jul 2020 13:10:43 +0200
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     1
      memory:  512Mi
    Requests:
      cpu:     250m
      memory:  256Mi
    Environment:
      VAULT_LOCAL_CONFIG:  {"listener":{"tcp":{"address":"0.0.0.0:8200","tls_cert_file":"/vault/tls/server.crt","tls_key_file":"/vault/tls/server.key"}},"storage":{"file":{"path":"${ .Env.VAULT_STORAGE_FILE }"}},"telemetry":{"statsd_address":"localhost:9125"},"ui":true}
      POD_NAME:            vault-central-0 (v1:metadata.name)
      VAULT_LOG_LEVEL:     debug
      VAULT_STORAGE_FILE:  /vault/file
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from vault-token-blcgd (ro)
      /vault/config from vault-config (rw)
      /vault/file from vault-file (rw)
      /vault/tls from vault-tls (rw)
Containers:
  vault:
    Container ID:  docker://ef9e5d924a7713d5f7aac2c4611ad79e60503ac39e8ee62bc28d2e83976138a2
    Image:         vault:1.4.3
    Image ID:      docker-pullable://vault@sha256:be3a63905b4d78d6e7c24fc9aacb30621d7a8a1ad8e8baff286a4ba110db16e8
    Ports:         8200/TCP, 8201/TCP
    Host Ports:    0/TCP, 0/TCP
    Args:
      server
    State:          Running
      Started:      Tue, 14 Jul 2020 13:10:44 +0200
    Ready:          False
    Restart Count:  0
    Limits:
      cpu:     1
      memory:  512Mi
    Requests:
      cpu:      250m
      memory:   256Mi
    Liveness:   http-get https://:api-port/v1/sys/init delay=0s timeout=1s period=10s #success=1 #failure=3
    Readiness:  http-get https://:api-port/v1/sys/health%3Fstandbyok=true&perfstandbyok=true delay=0s timeout=1s period=5s #success=1 #failure=2
    Environment:
      VAULT_LOG_LEVEL:     debug
      VAULT_STORAGE_FILE:  /vault/file
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from vault-token-blcgd (ro)
      /vault/config from vault-config (rw)
      /vault/file from vault-file (rw)
      /vault/tls from vault-tls (rw)
  bank-vaults:
    Container ID:  docker://119b9e104e8d6405c8b405412121de145214d481913d334ec71b0eac5992e6ab
    Image:         banzaicloud/bank-vaults:latest
    Image ID:      docker-pullable://banzaicloud/bank-vaults@sha256:5090ef735a295644aa7753c2f94dd4a65af940f8ae69bbc7828c665dcde20edd
    Port:          9091/TCP
    Host Port:     0/TCP
    Command:
      bank-vaults
      unseal
      --init
    Args:
      --pre-flight-checks
      true
      --mode
      k8s
      --k8s-secret-namespace
      system-vault-central
      --k8s-secret-name
      vault-central-unseal-keys
      --k8s-secret-labels
      app.kubernetes.io/name=vault,vault_cr=vault-central
    State:          Running
      Started:      Tue, 14 Jul 2020 13:10:45 +0200
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     200m
      memory:  128Mi
    Requests:
      cpu:     100m
      memory:  64Mi
    Environment:
      POD_NAME:      vault-central-0 (v1:metadata.name)
      VAULT_ADDR:    https://127.0.0.1:8200
      VAULT_CACERT:  /vault/tls/ca.crt
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from vault-token-blcgd (ro)
      /vault/tls from vault-tls (rw)
  prometheus-exporter:
    Container ID:  docker://dd604789e253c7181b3ba42201716e8b6b19c8b0c81a9f4940d4cd76c53a0849
    Image:         prom/statsd-exporter:latest
    Image ID:      docker-pullable://prom/statsd-exporter@sha256:44beed7ad8383328effa0cee14987769dd8137685b75e3297dacbfc5bea103c4
    Ports:         9125/UDP, 9102/TCP
    Host Ports:    0/UDP, 0/TCP
    Args:
      --statsd.mapping-config=/tmp/statsd-mapping.conf
    State:          Running
      Started:      Tue, 14 Jul 2020 13:10:45 +0200
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     200m
      memory:  128Mi
    Requests:
      cpu:        100m
      memory:     64Mi
    Environment:  <none>
    Mounts:
      /tmp/ from statsd-mapping (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from vault-token-blcgd (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  statsd-mapping:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      vault-central-statsd-mapping
    Optional:  false
  vault-config:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     Memory
    SizeLimit:  1Mi
  vault-file:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  vault-file
    ReadOnly:   false
  vault-tls:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  vault-central-tls
    Optional:    false
  vault-token-blcgd:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  vault-token-blcgd
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason     Age                       From                                            Message
  ----     ------     ----                      ----                                            -------
  Warning  Unhealthy  2m59s (x2641 over 3h42m)  kubelet, worker-prod-01.kube-prod.banana.systems  Readiness probe failed: HTTP probe failed with statuscode: 503
```

Vault operator:
```
$ kubectl describe -n system-vault-operator pod vault-operator-5fc487b958-6vw6w
Name:         vault-operator-5fc487b958-6vw6w
Namespace:    system-vault-operator
Priority:     0
Node:         worker-prod-01.kube-prod.banana.systems/10.202.1.30
Start Time:   Tue, 14 Jul 2020 13:09:35 +0200
Labels:       app.kubernetes.io/instance=vault-operator
              app.kubernetes.io/name=vault-operator
              pod-template-hash=5fc487b958
Annotations:  cni.projectcalico.org/podIP: 10.122.3.149/32
              cni.projectcalico.org/podIPs: 10.122.3.149/32
Status:       Running
IP:           10.122.3.149
IPs:
  IP:           10.122.3.149
Controlled By:  ReplicaSet/vault-operator-5fc487b958
Containers:
  vault-operator:
    Container ID:  docker://0a9df5a06bc541f841bcfc5ed86067c45dd3bb74cff2e3824ae19e33f42b1d06
    Image:         banzaicloud/vault-operator:1.3.3
    Image ID:      docker-pullable://banzaicloud/vault-operator@sha256:bf9fd4063892f9104f8a1723c0c02b3bc4ba15c6592159205c290992b7f49a82
    Ports:         8080/TCP, 8383/TCP
    Host Ports:    0/TCP, 0/TCP
    Command:
      vault-operator
      -sync_period
      1m
    State:          Running
      Started:      Tue, 14 Jul 2020 13:09:37 +0200
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     100m
      memory:  256Mi
    Requests:
      cpu:      100m
      memory:   128Mi
    Liveness:   http-get http://:8080/ delay=60s timeout=1s period=10s #success=1 #failure=3
    Readiness:  http-get http://:8080/ready delay=0s timeout=1s period=10s #success=1 #failure=3
    Environment:
      WATCH_NAMESPACE:
      POD_NAME:            vault-operator-5fc487b958-6vw6w (v1:metadata.name)
      OPERATOR_NAME:       vault-operator
      OPERATOR_LOG_LEVEL:  debug
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from vault-operator-token-7d5qj (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  vault-operator-token-7d5qj:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  vault-operator-token-7d5qj
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:          <none>
```

Any ideas what is wrong here? Thank you in advance for any help.

/kind bug

**Comment by bonifaido** [comment]
Have you cleaned up all resources of a possible previous Vault instance? Like PVCs?

**Comment by TomaszKlosinski** [comment]
> Have you cleaned up all resources of a possible previous Vault instance? Like PVCs?

Hi, I actually did remove all the resources and PVCs, but I found what is the problem - since I've used "Retain reclaim policy" for my StorageClass (I didn't want to lose data), I was actually getting the same data under new PVCs. So I had to first clean up the old data and then re-create the Vault instance. Thanks for pointing in the right direction.
